src/app/games/hive/HiveState.ts

```diff
@@ -154,31 +195,14 @@ export class HiveState extends FreeHexagonalGameState<HivePieceStack> implements
         }
     }
 
-    public isEmpty(pieces: HivePieceStack): boolean {
-        return pieces.isEmpty();
-    }
-
-    public override setAt(coord: Coord, pieces: HivePieceStack): void {
-        for (const player of Player.PLAYERS) {
-            // If there was a queen bee here, we remove it from the cache
-            if (this.queenBees.get(player).equalsValue(coord)) {
-                this.queenBees.delete(player);
-            }
-        }
-        for (const piece of pieces.pieces) {
-            // Add any queen added here to the cache
-            if (piece.kind === 'QueenBee') {
-                this.queenBees.put(piece.owner, coord);
-            }
-        }
-
-        super.setAt(coord, pieces);
-    }
-
     public queenBeeLocation(player: Player): MGPOptional<Coord> {
         return this.queenBees.get(player);
     }
 
+    public hasQueenBeeOnBoard(player: Player): boolean {
```

Comment:
is ?

Reply:
l'état *a* la reine sur le plateau
Ou alors : state.queenBeeIsOnBoard()
Tu préfère quoi ?

Reply:
nah aller, gardons l'originale ! TPR.